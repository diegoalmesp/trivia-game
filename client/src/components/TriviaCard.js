import React from 'react'
import PropTypes from 'prop-types'

const TriviaCard = ({
  trivia: {
    correct_answer,
    incorrect_answers,
    category,
    question
  }, onResponse, children}) => {
  const responses = [correct_answer, ...incorrect_answers].sort()
  return (
    <div className="trivia-page">
      <h3>{category}</h3>

      <div>
        <div className="trivia-question-container">
          <p className="text-monospace">{question}</p>
        </div>

        {children}
      </div>

      <div className="trivia-button-container">
        {responses.map((response, i) => 
          <button
            key={i}
            type="button"
            className="btn btn-info btn-lg text-uppercase"
            onClick={() => {onResponse(response)}}>
            {response}
          </button>
        )}
      </div>
    </div>
  )
}

TriviaCard.propTypes = {
  trivia: PropTypes.shape({
    category: PropTypes.string.isRequired,
    correct_answer: PropTypes.string.isRequired,
    incorrect_answers: PropTypes.array.isRequired,
    question: PropTypes.string.isRequired,
    type: PropTypes.string
  })
}

export default TriviaCard
