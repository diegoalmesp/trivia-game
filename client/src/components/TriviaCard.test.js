import React from 'react'
import renderer from 'react-test-renderer';
import { mount } from 'enzyme'

import TriviaCard from './TriviaCard'

describe('<TriviaCard />', () => {
  let wrapper

  const trivia = {
    correct_answer: 'True',
    incorrect_answers: ['False'],
    category: 'category',
    question: 'question'
  }

  describe('component rendering', () => {
    beforeEach(() => {
      wrapper = mount(<TriviaCard trivia={trivia} />)
    })

    it('renders correctly', () => {
      const card = renderer
        .create(<TriviaCard trivia={trivia} />)
        .toJSON()
      expect(card).toMatchSnapshot()
    })

    it('shoud render a h3 tag with the content `category`', () => {
      expect(
          wrapper.find('h3').first().length
      ).toBe(1)

      expect(
        wrapper.find('h3').first().text()
      ).toEqual('category')
    })

    it('shoud render a p tag with the content `question`', () => {
      expect(
          wrapper.find('div.trivia-question-container p').first().length
      ).toBe(1)

      expect(
        wrapper.find('div.trivia-question-container p').first().text()
      ).toEqual('question')
    })

    it('should render 2 buttons to answer', () => {
      expect(
        wrapper.find('div.trivia-button-container button').length
      ).toBe(2)
    })
  })

})