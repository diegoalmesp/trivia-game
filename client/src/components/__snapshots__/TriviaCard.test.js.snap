// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TriviaCard /> component rendering renders correctly 1`] = `
<div
  className="trivia-page"
>
  <h3>
    category
  </h3>
  <div>
    <div
      className="trivia-question-container"
    >
      <p>
        question
      </p>
    </div>
  </div>
  <div
    className="trivia-button-container"
  >
    <button
      className="btn btn-info btn-lg text-uppercase"
      onClick={[Function]}
      type="button"
    >
      False
    </button>
    <button
      className="btn btn-info btn-lg text-uppercase"
      onClick={[Function]}
      type="button"
    >
      True
    </button>
  </div>
</div>
`;
